src/conn_sock.c

```diff
@@ -188,7 +202,7 @@ static char *bind_unix_socket(char *socket_relative_name, int sock_type, mode_t
 	if (fchmod(socket_fd, perms))
 		pexitf("Failed to change socket permissions %s", sock_fullpath);
 
-	if (chdir(opt_socket_path) == -1)
+	if (do_chdir && chdir(opt_socket_path) == -1)
```

Comment:
Could we use the same trick with O_PATH I've used for Podman so we don't need a new flag and avoid using chdir?

Reply:
I think that's a good trick for the `do_chdir` case, but we still need this PR to maintain backwards compatibility with podman (that truncates the path once it gets long enough)

Reply:
acutally, that's a good trick for both cases, but we still will need to truncate the path sometimes, thus still need this flag

Reply:
don't we need to teach podman how to use the new flag though?  How will it work with older versions?

Reply:
yeah new podman will use this flag to be able to fix https://github.com/containers/conmon/issues/213
#213 will not be able to be fixed with older versions of podman

Reply:
basically, there's no backwards compatible way to fix #213, so we need to gate the fix with a new flag

Reply:
I still don't understand why we need a new flag.  Can't we fix the unlink with?

```diff
- 	if (unlink(attach_addr.sun_path) == -1 && errno != ENOENT)
+ 	if (unlink(attach_sock_path) == -1 && errno != ENOENT)
```

Reply:
nevermind my last comment.

For backward compatibility could we bump the API version and use the full path version only when the API version is 2?

Reply:
jw, is there a reason you're opposed to a flag here? we could use api-version here, but it seems like a very small change to bump the conmon major version. and if we did, we'd still need a change in podman to actually absorb the change, which is no different than a flag

Reply:
the main reason is that I'd prefer if possible to avoid `chdir` as it affects the entire process

Reply:
okay I dropped chdir but kept a flag. Now we gate the behavior of truncating the path on a flag, but regardless don't  chdir

Reply:
thanks!